pydantic/main.py

```diff
@@ -541,6 +541,7 @@ def model_json_schema(
         cls,
         by_alias: bool = True,
         ref_template: str = DEFAULT_REF_TEMPLATE,
+        union_format: Literal['any_of', 'primitive_type_array'] = 'any_of',
```

Comment:
Adding extra parameters like this is fine, but can be annoying for users subclassing `BaseModel` and overriding the `model_json_schema()` method, as they will need to update the definition.

One alternative is to have a single parameter (e.g. `TypedDict`) holding all the configuration, but it adds complexity on the caller side:

```python
Model.model_json_schema(settings={'union_format': ...})
```

And overriding the method isn't really a robust solution anyway.

However, we should definitely make these keyword only. This will be a breaking change, so either we do it know but make sure to properly announce it, or wait for V3 (and use deprecated overloads until then).

Reply:
Can you make `union_format` keyword-only now, and leave the rest for later? I guess that would also then not be breaking?

Reply:
:+1: , created https://github.com/pydantic/pydantic/issues/12332 as well.